user-interface/src/search-results/SearchResults.tsx

```diff
@@ -40,11 +40,20 @@ export type SearchResultsProps = {
 };
 
 export function SearchResults(props: SearchResultsProps) {
-  const { id, onStartSearching, onEndSearching } = props;
+  const {
+    id,
+    searchPredicate: searchPredicateProp,
+    onStartSearching,
+    onEndSearching,
+    noResultsMessage: noResultsMessageProp,
+    header: Header,
+    row: Row,
+    ...otherProps
+  } = props;
```

Comment:
I would just like to say about this general practice -- This may may provide for less typing when referring to properties throughout our components, but it also muddies the waters in regards to name spacing.  It becomes unclear as to the difference between variables defined as native to the component and those fed in through properties.  I would like to perhaps keep this practice to a minimum and only for variables that really need it.

I suppose on the flip side of that, there is the need to pull certain properties out that you need passed down to subcomponents and sometimes those are native DOM attributes that we don't necessarily know about... so in that case it's good to de-structure props because you can exclude certain things from "otherProps" that you don't want added to subcomponents.